# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/examples/blob.rs
# Commit: 4aa1704827b1
# Full Hash: 4aa1704827b131756b1dac86994e6963a2ba144f
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2024-08-08 09:35:37
# Regressor Bug: 1903977
# File Overlap Count: 1
# Description:
#   Bug 1903977 - Add a blob tile pool. r=gfx-reviewers,gw
#   
#   The tile pool keeps a strong reference to all of its allocations and reuses them when their reference count gets down to one.
#   Jemalloc only uses thread-local arenas for small allocations (<496 bytes) so it does not matter what thread the blob tiles which are typically much larger, are deallocated in.
#   
# ==============================================================================

diff -r b6440b259f94 -r 4aa1704827b1 gfx/wr/examples/blob.rs
--- a/gfx/wr/examples/blob.rs	Wed Aug 07 17:42:25 2024 +0000
+++ b/gfx/wr/examples/blob.rs	Wed Aug 07 18:16:58 2024 +0000
@@ -18,6 +18,7 @@
 use std::sync::Arc;
 use webrender::api::{self, DisplayListBuilder, DocumentId, PipelineId, PrimitiveFlags};
 use webrender::api::{ColorF, CommonItemProperties, SpaceAndClipInfo, ImageDescriptorFlags};
+use webrender::api::BlobTilePool;
 use webrender::api::units::*;
 use webrender::render_api::*;
 use webrender::euclid::size2;
@@ -183,7 +184,8 @@
     fn rasterize(
         &mut self,
         requests: &[api::BlobImageParams],
-        _low_priority: bool
+        _low_priority: bool,
+        _tile_pool: &mut BlobTilePool,
     ) -> Vec<(api::BlobImageRequest, api::BlobImageResult)> {
         let requests: Vec<(&api::BlobImageParams, Arc<ImageRenderingCommands>)> = requests.into_iter().map(|params| {
             (params, Arc::clone(&self.image_cmds[&params.request.key]))